### ISSUE TITLE
Add support for sequential masks

### ISSUE THREAD

**Issue by magnumripper** [openwall/john#4016]
See also #2310

We should allow something like `--mask='?w?a;?a?w'` (similar to `--rules=':rule1;rule2'`) although when `?w` or `?W` is used I'm not sure whether we should apply each mask to every word, or apply first mask to every word and then rewind (in case of wordlist) and use next mask.

**Comment by solardiz** [comment]
We should also have a way to read the multiple masks from a john.conf section or from a text file. I already needed that functionality when having a mask generate many masks, outputting that to a text file, then grepping out masks impossible by password policy, and using the rest for a cracking run. I ended up using a shell `for` loop around `john`, and would prefer to also have this functionality built-in.

**Comment by magnumripper** [comment]
> We should also have a way to read the multiple masks from a john.conf section or from a text file.

Absolutely, that's #2310. IIRC Jim started working on that so I didn't, but then it somehow didn't end up in any code.

**Comment by solardiz** [comment]
Added this and especially #2310 to "Potentially 1.9.0-jumbo-2 material". Sounds like a high-level addition to me, which will hopefully not break already existing functionality even if initially implemented with a bug. My biggest concern is that current mask (number?) should be recorded in `.rec` files.
